include/motor-control/core/stepper_motor/motor_interrupt_handler.hpp

```diff
@@ -221,9 +221,36 @@ class MotorInterruptHandler {
         set_buffered_move(Move{});
     }
 
-    auto get_encoder_pulses() { return hardware.get_encoder_pulses(); }
+    auto get_encoder_pulses() {
+        if (get_encoder_overflow_status() == true) {
+            hardware.clear_encoder_SR();
```

Comment:
we should either be always using things like `clear_encoder_overflow_flag` or always be using things like `hardware.clear_encoder_SR`, not both.